lib/order_repository.rb

```diff
@@ -0,0 +1,31 @@
+require_relative 'order'
+
+class OrderRepository
+  # return array of Order object and the corresponding item name string
+  def all_with_item
+    sql = 'SELECT 
+              orders.id, orders.customer_name, order_date, item_id, items.name
+            FROM orders
+            JOIN items ON orders.item_id = items.id;'
+    result_set = DatabaseConnection.exec_params(sql, [])
+    orders_with_item = []
+
+    result_set.each do |record|
+      order = Order.new
+      order.id = record['id'].to_i
+      order.customer_name = record['customer_name']
+      order.order_date = record['order_date']
+      order.item_id = record['item_id'].to_i
+      orders_with_item << [order, record['name']]
```

Comment:
I am wondering why you push an array containing order and the name, and what is `record['name']` if not customer name? Having arrays inside an array might make your code more confusing in general, because accessing object properties with dot notation is much simpler (and readable! like, `dog.legs`) than accessing array index (`dog[0] `might be legs but I don't know that when I am reading the code!)

Reply:
You have an `@items` array inside your `Order` class so why not use that to store your `Items`?